_posts/2021-04-14-accessing-the-relay-store.markdown

```diff
@@ -0,0 +1,260 @@
+---
+layout: epic
+title: "A Cool Way to Access the Relay Store"
+date: 2021-04-14
+categories: [relay, graphql, react, redis, tooling]
+author: anna
+---
+
+Need to access the Relay store without using a mutation? Relay's `commitLocalUpdate` allows you to modify the relay
+store from inside any relay component. The solution is particularly well suited to situations that just require
+client-side updates and do not involved changes to server-side data.
+
+<!-- more -->
+
+```js
+commitLocalUpdate(relay.environment, (store) => {
+  const parentRecord = store.get(parentID)
+
+  if (parentRecord) {
+    const artworksConnection = ConnectionHandler.getConnection(
+      parentRecord,
+      "ArtworksMissingMetadata_partner_artworksConnection"
+    )
+    if (artworksConnection) {
+      ConnectionHandler.deleteNode(artworksConnection, id)
+    }
+  }
+})
+```
+
+## Quick Intro to Artsy x Relay x Me
+
+Relay is a GraphQL client library maintained by Facebook engineers and enables rapid client-side data fetching in
+React applications. [Artsy's adoption of Relay][why-does-artsy-use-relay] coincided with our move toward using
+React Native for our mobile work around 2016. I joined Artsy as an engineer in November of 2020 (after
+[transitioning to engineering from a non-technical role at the
+company][how-losing-my-way-helped-me-find-my-way-back].) When I joined, I was about a year into React development
+and completely new to Relay.
+
+I work on the Partner Experience (PX) team at Artsy. We build and maintain software used by our gallery and auction
+house partners to sell artwork on Artsy. Although Relay is not particularly new to Artsy, it’s relatively new to
+our team’s main repository, Volt. (Volt is Artsy’s CMS used by gallery partners to manage their presences on the
+platform.) A topic for another blog post, but Volt’s structure is worth noting here. Volt is a collection of mini
+React apps injected into HAML views——our way of incrementally converting the codebase to our new stack.
+
+Relay is one of the many new pieces of technology I've learned on the job here at Artsy. Relay’s biggest advantage
+in my eyes is how it tightly couples the client view and API call (in our case, to the GraphQL layer of our stack,
+which we call Metaphysics.) In addition to performance and other benefits, having the data fetch right on top of
+the React component creates a pretty seamless developer experience.
```

Comment:
Maybe @pepopowitz can help me out here... I'm tying to think of a more succinct way to describe this part about data fetching on top of a react component. My wordsmithing is failing... The way its put here is not incorrect, because there is indeed a fetch happening in the tree, but technically speaking what's we're doing is describing a data requirement which then gets resolved (and actually fetched) somewhere in the parent near the `QueryRenderer`. That's why the language often used leans heavily on "describing a component's needs" and "colocation", etc.

Reply:
I think "colocation" is a solid succinct description of it. Maybe something like...

"In addition to performance and other benefits, colocating a component with its data requirements creates a pretty seamless developer experience."

Reply:
YES love it thanks guys! And this is actually a helpful clarification for me to understand what's actually happening.